code/go/internal/validator/semantic/types.go

```diff
@@ -362,22 +372,41 @@ func listTransforms(fsys fspath.FS) ([]string, error) {
 
 }
 
-func listPipelineFiles(fsys fspath.FS, dataStream string) ([]pipelineFileMetadata, error) {
+func listPipelineFiles(fsys fspath.FS) ([]pipelineFileMetadata, error) {
 	var pipelineFileMetadatas []pipelineFileMetadata
 
-	ingestPipelineDir := path.Join(dataStreamDir, dataStream, "elasticsearch", "ingest_pipeline")
-	pipelineFiles, err := readPipelinesFolder(fsys, ingestPipelineDir)
-	if err != nil {
-		return nil, fmt.Errorf("cannot read pipeline files from integration package: %w", err)
+	type pipelineDirMetadata struct {
+		dir        string
+		dataStream string
 	}
 
-	for _, file := range pipelineFiles {
-		pipelineFileMetadatas = append(pipelineFileMetadatas, pipelineFileMetadata{
-			filePath:     file,
-			fullFilePath: fsys.Path(file),
-			dataStream:   dataStream,
+	dirs := []pipelineDirMetadata{{dir: ""}}
+
+	dataStreams, err := listDataStreams(fsys)
+	if err != nil {
+		return nil, specerrors.ValidationErrors{specerrors.NewStructuredError(err, specerrors.UnassignedCode)}
+	}
+	for _, dataStream := range dataStreams {
+		dirs = append(dirs, pipelineDirMetadata{
+			dir:        path.Join(dataStreamDir, dataStream),
+			dataStream: dataStream,
 		})
 	}
 
+	for _, d := range dirs {
+		pipelinePath := path.Join(d.dir, "elasticsearch", "ingest_pipeline")
+		pipelineFiles, err := readPipelinesFolder(fsys, pipelinePath)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read pipeline files from integration package: %w", err)
+		}
+		for _, file := range pipelineFiles {
+			pipelineFileMetadatas = append(pipelineFileMetadatas, pipelineFileMetadata{
+				filePath:     file,
+				fullFilePath: fsys.Path(file),
+				dataStream:   d.dataStream,
```

Comment:
Is this `dataStream` field form `pipelineFileMetadata` struct used ? 
Taking a look at #1010 and this PR, it looks like it is not. Could you check it? If not, should it be kept or used somewhere?